### PR TITLE
Riaz Ahmed - Test Assignment

### PR DESCRIPTION
**Pull request by riazahmed01**
I dont think I did the github actions properly. I am including some screenshots

jest:
image

image

cypress e2e:
image

### CODE REVIEW

**Comment by riazahmed01** [comment]
I apologize if i set off multiple emails from workflow fail.

File: tests/UnitTests/Tabs.test.js [hunk 1]
```diff
@@ -1,28 +0,0 @@
-import React from 'react';
-import { render, fireEvent } from '@testing-library/react'; 
-import Tabs from '@/components/Tabs/Tabs';
-
-test('it should change the active tab when a button is clicked', () => {
```
Comment (navteniev) [inline tests/UnitTests/Tabs.test.js:5]: Why was this test deleted?

File: src/components/SearchBar/SearchBar.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      const searchButton = screen.getByTestId('search-button');
+  
+      fireEvent.change(searchInput, { target: { value: 'example' } });
+      fireEvent.click(searchButton);
+  
+      expect(pushMock).toHaveBeenCalledWith('/explore/?search=example');
+    });
+
+    //for empty search, since we don't have any input restriction
+    it('should update searchQuery on button click', () => {
```
Comment (navteniev) [inline src/components/SearchBar/SearchBar.test.js:31]: This seems to be the same test as above and does not test an error condition necessarily

File: cypress/e2e/spec.cy.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+describe('template spec', () => {
```
Comment (navteniev) [inline cypress/e2e/spec.cy.js:1]: Seems to be an artifact of setting up cypress not part of the test suit.

File: src/app/explore/explore.test.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  };
+});
+
+jest.mock('next/navigation', () => ({
+  useRouter: jest.fn(),
+  useSearchParams: jest.fn(),
+}));
+
+describe('Explore Page Integration Test', () => {
+  it('should send search value to API and get a response', async () => {
```
Comment (navteniev) [inline src/app/explore/explore.test.js:20]: This seems to be testing the explore page in insolation from the API call it makes rather then testing the integration between the two.

There is also no test for an error condition.

**Review by navteniev — COMMENTED** [review]
Unit Tests: 2/2 pts
Integration: 1/2pts missing error condition test
e2e: 1/1pts
CI: 4/4 pts there were no jobs executed likely because the branch conditions were incorrect. It would have been useful to check these using act.
PR: 1/1 pts

Final: 9/10 pts

**Pull request closed without merging**